Describe how to install Chef Infra Server using Chef Automate (#2851)
chef/automate: components/automate-chef-io/content/docs/infra-server.md

+++
title = "Install Chef Infra Server With Automate"
description = "Use Automate to install Chef Infra Server"
date = 2020-02-11T14:24:00-08:00
weight = 20
draft = false
bref = ""
toc = true
[menu]
  [menu.docs]
    parent = "get_started"
    weight = 60
+++

## Overview
Use Chef Automate to install Chef Infra Server.

## System requirements and prerequisites
### Hardware
See the [hardware requirements for Chef Infra Server](https://docs.chef.io/server_overview.html) for guidance on memory and number of CPUs for a standalone Chef Infra Server installation.

For a single-host installation that contains Chef Infra Server and Chef Automate, start

> with

Author: Suggestions on disk/memory for Automate + Infra Server? Aside from the disk reqs, these are just the reqs for Automate so far.

instructions and guidance on using and managing your Chef Infra Server.

## System requirements and prerequisites

Before beginning your installation, check the [System Requirements]({{< relref "system-requirements.md" >}}) for Automate, and ensure that the `chef-automate` command line tool is installed.

To download the `chef-automate` command line tool, run the following command in your command line interface:

```shell
curl https://packages.chef.io/files/current/latest/chef-automate-cli/chef-automate_linux_amd64.zip | gunzip - > chef-automate && chmod +x chef-automate
```

## Install Chef Automate and Chef Infra Server on the same host

Use either a command line interface or a configuration file to install Chef Automate and Chef Infra Server on the same host. Run the commands as the superuser or use `sudo` at the start of each command.

### Hardware Requirements

For a single-host installation that contains Chef Infra Server and Chef Automate, we recommend the following memory and vCPU minimums:

* up to 200 managed nodes: 8GB RAM, 2 vCPUs
* between 200 and 500 managed nodes: 30GB RAM, 4 vCPUs
* between 500 and 5000 managed nodes: 61GB RAM, 8vCPUs

> A single-host installation that contains Chef Infra Server and Chef Automate will require 80 GB of disk space plus 2 MB of disk space per managed node that is available to `/hab`.

Reviewer: @yzl Also, I worked on rephrasing this line, but I was unsure exactly what `/hab` referred to. Can you please clarify what `/hab` refers to? Thanks! :)

Author: `/hab` is the default directory to which habitat installs software.

```shell
sudo chef-automate deploy --product automate --product chef-server
```

### Via Configuration

To install Chef Automate and Chef Infra Server on the same host via configuration will still require a command line interface.

1. First, generate a skeleton configuration file by running this command:

    ```shell
      sudo chef-automate init-config
    ```

1. Add a stanza to the configuration file to deploy Automate and Chef Infra Server:

    ```toml
      [deployment.v1.svc]
      products=["automate", "chef-server"]
    ```

1. Make any other configuration changes desired.

1. Run the `chef-automate deploy` command with your configuration file:

    ```shell
      sudo chef-automate deploy config.toml
    ```

## Install standalone Chef Infra Server using Chef Automate

You can use either a command line option or a configuration file to install Chef Infra Server via Chef Automate. Run the commands as the superuser or use `sudo` at the start of each command.

### Hardware Requirements

For a standalone Chef Infra Server installation, see the [hardware requirements for Chef Infra Server](https://docs.chef.io/install_server_pre/) for guidance on memory and number of CPUs.

> ### Via command line

Reviewer: @yzl This series of steps involves a configuration file despite being labeled installation via command line. Is there something that needs to be amended in this instruction?

> ### Installation Via Command Line

Reviewer: @yzl This series of steps involves a configuration file despite being labeled installation via command line. Is there something that needs to be amended in this instruction?

Author: @mjingle The distinction really is whether the product is specified via the config file or the `--product` flag. I changed the section headings to be explicit about this.

To download the `chef-automate` command line tool, run the following command in your command line interface:

```shell
curl https://packages.chef.io/files/current/latest/chef-automate-cli/chef-automate_linux_amd64.zip | gunzip - > chef-automate && chmod +x chef-automate
```

## Install Chef Automate and Chef Infra Server on the same host

Use either a command line interface or a configuration file to install Chef Automate and Chef Infra Server on the same host. Run the commands as the superuser or use `sudo` at the start of each command.

### Hardware Requirements

For a single-host installation that contains Chef Infra Server and Chef Automate, we recommend the following memory and vCPU minimums:

* up to 200 managed nodes: 8GB RAM, 2 vCPUs
* between 200 and 500 managed nodes: 30GB RAM, 4 vCPUs
* between 500 and 5000 managed nodes: 61GB RAM, 8vCPUs

A single-host installation that contains Chef Infra Server and Chef Automate will require a `/hab` directory that contains 80 GB of disk space for software artifacts plus 2 MB of disk space per managed node.

> ### Specify Products From the Command Line

Reviewer: @yzl  Is there a distinction that needs to be made between "specify" in the header and "install" used in the content below? Asking to determine if wording should change through the rest of this section.

Author: @mjingle "Specify" is the mechanism. "Install" is the job to be done. We want to use "specify" in the header because the job to be done is always "install Automate", but users will chose different mechanisms based on their needs.

Reviewer 2: Simple present, please. "...Chef Automate requires..."

> To install Chef Automate and Chef Infra Server on the same host, run this command:

Reviewer 2: ```suggestion
Install Chef Automate and Chef Infra Server on the same host with this command:
```

```shell
sudo chef-automate deploy --product automate --product chef-server
```

### Specify Products Via the Configuration

> To install Chef Automate and Chef Infra Server on the same host via configuration still requires a command line interface.

Reviewer 2: I don't understand this sentence. Do you mean:
```suggestion
Installing Chef Automate and Chef Infra Server on the same host using a configuration file also requires the use of the Chef Automate CLI.
```

1. First, generate a skeleton configuration file by running this command:

    ```shell
      sudo chef-automate init-config
    ```

1. Add a stanza to the configuration file to deploy Automate and Chef Infra Server:

    ```toml
      [deployment.v1.svc]
      products=["automate", "chef-server"]
    ```

1. Make any other configuration changes desired.

1. Run the `chef-automate deploy` command with your configuration file:

    ```shell
      sudo chef-automate deploy config.toml
    ```

## Install standalone Chef Infra Server using Chef Automate

> You can use either a command line option or a configuration file to install Chef Infra Server via Chef Automate. Run the commands as the superuser or use `sudo` at the start of each command.

Reviewer 2: ```suggestion
You can install Chef Infra Server through Chef Automate using either the command line or with a configuration file. Run the commands as the superuser or use `sudo` at the start of each command.
```

### Hardware Requirements

For a standalone Chef Infra Server installation, see the [hardware requirements for Chef Infra Server](https://docs.chef.io/install_server_pre/) for guidance on memory and number of CPUs.

> ### Specify Product on the Command Line

Reviewer 2: The duplicate headings won't pass Markdown linting. Add short name of process/thing installed to fix. Avoid duplicating the titles in the install server doc in chef-web-docs.
